packages/addon-dev/src/rollup-app-reexports.ts

```diff
@@ -25,21 +26,21 @@ export default function appReexports(opts: {
           this.emitFile({
             type: 'asset',
             fileName: `_app_/${appFilename}`,
-            source: `export { default } from "${pkg.name}/${addonFilename.slice(
-              0,
-              -extname(addonFilename).length
-            )}";\n`,
+            source: `export ${opts.defaultExport} from "${
+              pkg.name
+            }/${addonFilename.slice(0, -extname(addonFilename).length)}";\n`,
           });
         }
       }
       let originalAppJS = pkg['ember-addon']?.['app-js'];
-
-      let hasChanges = JSON.stringify(originalAppJS) !== JSON.stringify(appJS);
+      let newAppJs = Object.assign({}, originalAppJS || {}, appJS);
```

Comment:
This never removes deleted things from app-js.

Instead of trying to make it possible to layer multiple copies of this plugin together, I think it would be better to offer an API so that it can generate a correct app-js in one pass.